example-app/SantokuApp/src/bases/ui/common/StyledFlatList.tsx

```diff
@@ -0,0 +1,23 @@
+import {BoxProps, createBox} from '@shopify/restyle';
+import React from 'react';
+import {FlatList, FlatListProps} from 'react-native';
+
+import {RestyleTheme} from '../theme/restyleTheme';
+
+// createBoxでは、FLatListPropsのジェネリクスの型定義ができない
+// https://github.com/Shopify/restyle/issues/166
+export const BaseFlatList = createBox<RestyleTheme, FlatListProps<any>>(FlatList);
+
+type StyledFlatListProps<T> = BoxProps<RestyleTheme> & FlatListProps<T>;
+const StyledFlatListComponent = <T extends unknown>(
+  {children, ...rest}: StyledFlatListProps<T>,
+  ref: React.ForwardedRef<FlatList>,
+) => (
+  <BaseFlatList ref={ref} {...rest}>
+    {children}
+  </BaseFlatList>
+);
+
+export const StyledFlatList = React.forwardRef(StyledFlatListComponent) as <T>(
+  props: BoxProps<RestyleTheme> & FlatListProps<T> & {ref?: React.ForwardedRef<FlatList>},
+) => ReturnType<typeof StyledFlatListComponent>;
```

Comment:
Wouldn't it be equivalent to the following code?
```typescript
type StyledFlatListProps<T> = BoxProps<RestyleTheme> & FlatListProps<T>;
function StyledFlatListComponent<T>({children, ...rest}: StyledFlatListProps<T>, ref: React.ForwardedRef<FlatList>) {
  return (
    <BaseFlatList ref={ref} {...rest}>
      {children}
    </BaseFlatList>
  );
}

export const StyledFlatList = React.forwardRef(StyledFlatListComponent);
```

Reply:
The above method would result in a type error in the `renderItem` of the component that uses StyledFlatList.

https://fettblog.eu/typescript-react-generic-forward-refs/

Reply:
Thank you. I learned a lot.